documentation/src/docs/basic-reader-and-writer.md

```diff
@@ -52,178 +48,167 @@ public void run(String routingKey, String message) {
 a developer to control various behaviors of the Stream.  All configuration
-objects in Pravega use a builder pattern for construction.  There are really two
-important configuration items related to streams: Retention Policy and Scaling
-Policy.  
-
-Retention Policy allows the developer to control how long data is kept in a
-Stream before it is deleted.  S/he can specify data should be kept for a certain
-period of time (ideal for situations like regulatory compliance that mandate
-certain retention periods) or to retain data until a certain number of bytes
-have been consumed.  At the moment, Retention Policy is not completely
-implemented.  By default, the RetentionPolicy is set as "unlimited" meaning, data
-will not be removed from the Stream.
-
-Scaling Policy is the way developers configure a Stream to take advantage
-Pravega's auto-scaling feature.  In line 6, we use a fixed policy, meaning the
-Stream is configured with the given number of Stream Segments and that won't
-change.  The other options are to scale by a given number of Events per second
-or a given number of Kilobytes per second.  In these two policies, the developer
-specifies a target rate, a scaling factor and a minimum number of Segments.  The
-target rate is straight forward, if ingest rate exceeds a certain number of
-Events or Kilobytes of data for a sustained period of time, Pravega will attempt
-to add new Stream Segments to the Stream.  If the rate drops below that
-threshold for a sustained period of time, Pravega will attempt to merge adjacent
-Stream Segments.  The scaling factor is a setting on the Scaling Policy that
-determines how many Stream Segments should be added when the target rate (of
-Events or Kilobytes) is exceeded.  The minimum number of Segments is a factor
-that sets the minimum degree of read parallelism to be maintained; if this value
-is set at 3, for example, there will always be 3 Stream Segments available on
-the Stream.  Currently, this property is effective only when the stream is
+objects in Pravega use a builder pattern for construction. **Retention Policy** and **Scaling
+Policy** are the two important configuration items related to streams.   
+
+**Retention Policy:**  Pravega supports the following two types of Retentions:
+
+ - **Time based Retention**: It allows the developer to control how long the data is kept in a Stream before it is deleted. The developer can specify the time limit (milliseconds) to keep the data for a certain period of time (ideal for situations like regulatory compliance that mandate
+certain retention periods).
+ - **Size based Retention**: Retains the data until a certain number of bytes
+have been consumed. It is based on the total size of the data in the stream in bytes.
+
+**Scaling Policy:** When a Stream is created, it is configured with a Scaling Policy that determines how a Stream handles the varying changes in its load. Developers configure a Stream to take advantage
```

Comment:
I don't think that we have to repeat exactly the  same that is already in `Concepts.md`. It would be better to just introduce the idea and then refer to the `Concepts.md` page via a link. This page is about a practical example and it is good to stick to that idea. Otherwise the reader may loose focus.